app/src/pages/authoring/[resourceType]/[id].tsx

```diff
@@ -16,11 +16,28 @@ export default function ResourceAuthoringPage() {
 
   const ctx = trpc.useContext();
 
-  const resourceQuery = trpc.draft.getDraftById.useQuery({
+  const { data: resource } = trpc.draft.getDraftById.useQuery({
     id: id as string,
     resourceType: resourceType as ArtifactResourceType
   });
 
+  // useEffect to check if it has an identifier and url
+  useEffect(() => {
+    if (resource?.url) {
+      setUrl(resource.url);
+    }
+    if (resource?.identifier) {
+      const cmsIdentifier = resource.identifier.find(
+        identifier => identifier.system === 'http://hl7.org/fhir/cqi/ecqm/Measure/Identifier/cms'
+      );
+      if (cmsIdentifier?.value) {
+        setIdentifier(cmsIdentifier.value);
+      } else if (resource.identifier[0].value) {
+        setIdentifier(resource.identifier[0].value);
+      }
+    }
+  }, [resource?.url, resource?.identifier]);
```

Comment:
Maybe let's just put `resource` as the dependency, assuming we will have more inputs in the future, and I never really like using `?` in useEffect dependencies